SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -576,11 +584,21 @@ void USpatialActorChannel::OnCreateEntityResponse(const Worker_CreateEntityRespo
 
 	if (Op.status_code != WORKER_STATUS_CODE_SUCCESS)
 	{
-		UE_LOG(LogSpatialActorChannel, Error, TEXT("Failed to create entity for actor %s: %s"), *Actor->GetName(), UTF8_TO_TCHAR(Op.message));
-		Sender->SendCreateEntityRequest(this, GetPlayerWorkerId());
+		// UNR-630 - Temporary hack to avoid failure to create entities due to timeout on large maps
+		if (Op.status_code == WORKER_STATUS_CODE_TIMEOUT)
+		{
+			UE_LOG(LogSpatialActorChannel, Error, TEXT("Failed to create entity for actor %s due to timeout.  Retrying now..."), *Actor->GetName());
+			Sender->SendCreateEntityRequest(this, GetPlayerWorkerId());
+		}
+		else
+		{
+			UE_LOG(LogSpatialActorChannel, Error, TEXT("Failed to create entity for actor %s: %s"), *Actor->GetName(), UTF8_TO_TCHAR(Op.message));
+		}
+
 		return;
 	}
-	UE_LOG(LogSpatialActorChannel, Verbose, TEXT("Created entity (%lld) for: %s."), EntityId, *Actor->GetName());
+
+	UE_LOG(LogSpatialActorChannel, Verbose, TEXT("Created entity (%lld) for: %s."), Op.entity_id, *Actor->GetName());
```

Comment:
Found a bug here in the UE_LOG.  Looks like a copy-paste from the above "reserve" function, where 'EntityId' was defined as being 'Op.entity_id'.  No such declaration was made in the "create" function.

Reply:
Not really a bug, `EntityId` member variable would have been in reserve because we don't create without it. But this works too.